cmd/libs/go2idl/conversion-gen/generators/conversion.go

```diff
@@ -638,18 +705,47 @@ func (g *genConversion) doStruct(inType, outType *types.Type, sw *generator.Snip
 			outMemberType = &copied
 		}
 
-		args := map[string]interface{}{
-			"inType":  inMemberType,
-			"outType": outMemberType,
-			"name":    inMember.Name,
+		args := argsFromType(inMemberType, outMemberType).With("name", inMember.Name)
+
+		// try a direct memory copy for any type that has exactly equivalent values
+		if g.useUnsafe.Equal(inMemberType, outMemberType) {
+			args = args.
+				With("Pointer", types.Ref("unsafe", "Pointer")).
+				With("SliceHeader", types.Ref("reflect", "SliceHeader"))
+			switch inMemberType.Kind {
+			case types.Pointer:
+				sw.Do("out.$.name$ = ($.outType|raw$)($.Pointer|raw$(in.$.name$))\n", args)
+				continue
+			case types.Map:
+				sw.Do("{\n", nil)
+				sw.Do("m := (*$.outType|raw$)($.Pointer|raw$(&in.$.name$))\n", args)
+				sw.Do("out.$.name$ = *m\n", args)
+				sw.Do("}\n", nil)
+				continue
+			case types.Slice:
+				sw.Do("{\n", nil)
+				sw.Do("outHdr := (*$.SliceHeader|raw$)($.Pointer|raw$(&out.$.name$))\n", args)
+				sw.Do("inHdr := (*$.SliceHeader|raw$)($.Pointer|raw$(&in.$.name$))\n", args)
+				sw.Do("*outHdr = *inHdr\n", nil)
+				sw.Do("}\n", nil)
+				continue
+			}
 		}
+
 		// check based on the top level name, not the underlying names
 		if function, ok := g.preexists(inMember.Type, outMember.Type); ok {
-			args["function"] = function
-			sw.Do("if err := $.function|raw$(&in.$.name$, &out.$.name$, s); err != nil {\n", args)
-			sw.Do("return err\n", nil)
-			sw.Do("}\n", nil)
-			continue
+			if isDrop(function.CommentLines) {
+				continue
+			}
+			// copy-only functions that are directly assignable can be inlined instead of invoked
+			if !isCopyOnly(function.CommentLines) || !g.isFastConversion(inMemberType, outMemberType) {
```

Comment:
I'm not sure I understand this - can you please clarify the idea here? And maybe given an example for this one?